Feature school controler

tabacha opened the pull request:
Fixes #9

Commit:
add school controler, #9

Commit:
add school controler, #9

jetwell reviewed (changes requested):
* The empty constructors in BaseEntity and School can be deleted (unless the framework requires them for some magic)
* The field $logger in SchoolController must be declared above the constructor (maybe also include PHPDoc '@var').
* The Swagger-Documentation is great. Is it possible to include PHPDocs (e.g. '@param' and '@return') in the documentation blocks too? Or would that cause problems?
* We haven't discussed code style issues such as camel case. In my opinion the variables 'singleresult' and 'postalcode' are easier to read if written in camel case: 'singleResult' and 'PostalCode'. If you prefer the current style then that's OK too.
* I fixed minor formatting issues (newlines and spaces).
* Your test method in SchoolControllerTest contains series of tests in a single method. Is is possible to break these down into multiple test methods? You could then give the methods descriptive names like 'testForbiddenForStudentAccount'. It's generally easier to maintain test suites with lots of small tests.

Commit:
Fixed whitespace issues.

tabacha committed:
Merge branch 'master' into feature_school_controler

Commit:
fixes, by php-cs-fixer

Commit:
seperate tests

Commit:
fix depreaction warning

Commit:
add more phpdoc and swagger documentation

tabacha commented:
> * The empty constructors in BaseEntity and School can be deleted (unless the framework requires them for some magic)

are required for doctrine

> * The field $logger in SchoolController must be declared above the constructor (maybe also include PHPDoc '@var').

done

> * The Swagger-Documentation is great. Is it possible to include PHPDocs (e.g. '@param' and '@return') in the documentation blocks too? Or would that cause problems?

we can mix them. I have tried to add phpdocs

> * We haven't discussed code style issues such as camel case. In my opinion the variables 'singleresult' and 'postalcode' are easier to read if written in camel case: 'singleResult' and 'PostalCode'. If you prefer the current style then that's OK too.

i like camelCase

> * I fixed minor formatting issues (newlines and spaces).

thank you

> * Your test method in SchoolControllerTest contains series of tests in a single method. Is is possible to break these down into multiple test methods? You could then give the methods descriptive names like 'testForbiddenForStudentAccount'. It's generally easier to maintain test suites with lots of small tests.

I have split it. But I have a problem with that: The tests are now depends on each other. I have learned, that every test schould work standalone. How do we deal about that? Sadly I found no posibility to load fixtures before each test.


BTW: You can use github feature to comment a code review directly in the code.

Commit:
fix php-cs (was ignored by the cache)

jetwell commented:
I would leave the tests as they are for now. It looks like the test file itself could be split into multiple files. Then the dependencies could be removed by introducing setups and teardowns for groups of similar tests. But we can do that when we next refactor the code.
